### PR TITLE
Feature/making progress and finishing quests

### PR DESCRIPTION
**Pull request by jekattack**
Added some Backend-Functionality:
- QuestItems now has status of STARTED, EXPIRED or DONE (created, when user starts quest. Previous name of the object was QuestStarted) 
When new Visit is created: 
- Checks, if there are STARTED quests for user
- Checks, if visit is relevant for quest
- Checks, if Quest now is DONE and adds points to score
- Else Visit is added as before
- Added some catches for Exceptions
- Minor changes

### CODE REVIEW

File: backend/src/test/java/com/github/jekattack/cornergame/game/quests/QuestServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import com.github.jekattack.cornergame.game.gamedata.questItem.QuestStatus;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.time.Instant;
+import java.util.*;
+
+class QuestServiceTest {
+
```
Comment (mysterix5) [inline backend/src/test/java/com/github/jekattack/cornergame/game/quests/QuestServiceTest.java:16]: some edge cases are left here i think, fail to create, maybe adding something double...

**Review by mysterix5 — APPROVED** [review]
spend a long time reading it but didnt find anything... 
lgtm!
